### PR TITLE
ref(replays): Refactor Replay Details Breadcrumbs list to use react-virtualized

### LINKED ISSUE #44165: Design: Include `Tags` section when the window is in portrait mode layout

### LINKED ISSUE #45842: Virtualize rendering of the Breadcrumbs areas inside the Replay Details page
We had a customer notice that playback of a replay was really choppy on their computer. The replay in question had 8,637 breadcrumbs, most coming from Redux, so the bodies of those breadcrumbs were not small either. This resulted in a lot of html on the page, re-rendering often during playback because of the row-highlighting and autoscroll.

Virtual render of this list, like with all the other lists, should smooth out playback and make things easier for the browser to render.

### PR DESCRIPTION
**Pull request by ryan953**
A few style changes along the way:
- The "portrait" aka "topbar" layout now shows tags. Previously Tags were hidden completely in this layout
- I dropped the rounded corners and margin around the breadcrumb items, they sit flush against each other and against the border.
- Before we showed 4 loading `<Placeholder>` rows, now we show a gray Placeholder over the whole section.
    - Without the margin/rounded corners & margin having individual items doesn't make sense. Now it matches with the other tabs better; console/network/etc.
    - Once the replayRecord is downloaded you'll see the "Start Recording" crumb, which seems to be replaced once everything else is done. But this is a good placeholder.
- Moved BreadcrumbItem into components, it's used in Timeline, UrlWalker, and the Breadcrumb List.
    - The list item added a wrapper around it, for reading from useReplayContext and some other hooks. 

Tested that things continue to work:
- auto scroll breadcrumbs when the video is playing
- text overflow inside breadcrumbs
    - was able to remove an extra `<div>` in each crumb item

| Window Orientation | Before | After |
| --- | --- | --- |
| Landscape (the usual) | widescreen list - before | widescreen list - after |
| Portrait | tall list - before | tall list - after |

Fixes https://github.com/getsentry/sentry/issues/45842
Fixes https://github.com/getsentry/sentry/issues/44165

### CODE REVIEW

File: static/app/utils/replays/hooks/useCurrentItemScroller.tsx
Comment (ryan953) [inline static/app/utils/replays/hooks/useCurrentItemScroller.tsx:1]: replaced by `useScrollToCurrentItem.tsx`

File: static/app/views/replays/detail/iconWrapper.tsx
Comment (ryan953) [inline static/app/views/replays/detail/iconWrapper.tsx:1]: This was copy+pasted into 3 of these files. All consolidated here now

File: static/app/views/replays/detail/layout/sidebarArea.tsx
Comment (ryan953) [inline static/app/views/replays/detail/layout/sidebarArea.tsx:1]: Make things more similar to focusArea.tsx

File: static/app/views/replays/detail/layout/index.tsx [hunk 1]
```diff
@@ -64,6 +62,14 @@ function ReplayLayout({layout = LayoutKey.topbar}: Props) {
     </ErrorBoundary>
   );
 
+  const sidebarArea = (
```
Comment (ryan953) [inline static/app/views/replays/detail/layout/index.tsx:65]: more similar to `const focusArea = ` which is just above here.

File: static/app/views/replays/detail/layout/index.tsx [hunk 2]
```diff
@@ -123,12 +129,6 @@ function ReplayLayout({layout = LayoutKey.topbar}: Props) {
   }
 
   // layout === 'topbar'
-  const crumbsWithTitle = (
-    <ErrorBoundary mini>
-      <Breadcrumbs showTitle />
-    </ErrorBoundary>
-  );
```
Comment (ryan953) [inline static/app/views/replays/detail/layout/index.tsx:130]: This is the version without Tags

Now we're using `sidebarArea` which has tags. See screen shots.

File: static/app/views/replays/detail/layout/sidebarArea.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import Breadcrumbs from 'sentry/views/replays/detail/breadcrumbs';
+import TagPanel from 'sentry/views/replays/detail/tagPanel';
+
+function SidebarArea() {
+  const {getParamValue} = useUrlParams('t_side', 'crumbs');
+  const {replay} = useReplayContext();
+
+  switch (getParamValue()) {
+    case 'tags':
+      return <TagPanel />;
```
Comment (billyvg) [inline static/app/views/replays/detail/layout/sidebarArea.tsx:12]: 🙏

**Review by billyvg — APPROVED** [review]
I clicked and scrolled around on a replay with a few hundred breadcrumbs and it feels pretty good to me!

**Pull request merged**